lanelet2_python/python_api/core.cpp

```diff
@@ -552,289 +664,505 @@ BOOST_PYTHON_MODULE(PYTHON_API_MODULE_NAME) {  // NOLINT
-                                  ConstLaneletWithStopLine{
-                                    std::move(lanelet), std::move(stopLine)
-                                  }
-                              );
-                           },
-          default_call_policies(),
-          (arg("lanelet"), arg("stopLine") = Optional<ConstLineString3d>{})))
-      .add_property("lanelet", &ConstLaneletWithStopLine::lanelet)
+           make_constructor(
+               +[](Lanelet lanelet, Optional<ConstLineString3d> stopLine) {
+                 return std::make_shared<ConstLaneletWithStopLine>(
+                     ConstLaneletWithStopLine{std::move(lanelet), std::move(stopLine)});
+               },
+               default_call_policies(), (arg("lanelet"), arg("stopLine") = Optional<ConstLineString3d>{})),
+           "Construct from a lanelet and a stop line")
+      .add_property("lanelet", &ConstLaneletWithStopLine::lanelet, "The lanelet")
       .add_property(
           "stopLine", +[](const ConstLaneletWithStopLine& self) { return optionalToObject(self.stopLine); },
           +[](ConstLaneletWithStopLine& self, const object& value) {
             self.stopLine = objectToOptional<LineString3d>(value);
-          });
+          },
+          "The stop line for the lanelet (can be None)");
 
   class_<LaneletWithStopLine>("LaneletWithStopLine", "A lanelet with a stopline", no_init)
-      .def("__init__",
-          make_constructor(+[](Lanelet lanelet, Optional<LineString3d> stopLine){
-                              return std::make_shared<LaneletWithStopLine>(
-                                  LaneletWithStopLine{
-                                    std::move(lanelet), std::move(stopLine)
-                                  }
-                              );
+      .def("__init__", make_constructor(
+                           +[](Lanelet lanelet, Optional<LineString3d> stopLine) {
+                             return std::make_shared<LaneletWithStopLine>(
+                                 LaneletWithStopLine{std::move(lanelet), std::move(stopLine)});
                            },
-          default_call_policies(),
-          (arg("lanelet"), arg("stopLine") = Optional<LineString3d>{})))
-      .add_property("lanelet", &LaneletWithStopLine::lanelet)
+                           default_call_policies(), (arg("lanelet"), arg("stopLine") = Optional<LineString3d>{})))
+      .add_property("lanelet", &LaneletWithStopLine::lanelet, "The lanelet")
       .add_property(
           "stopLine", +[](const LaneletWithStopLine& self) { return optionalToObject(self.stopLine); },
           +[](LaneletWithStopLine& self, const object& value) {
             self.stopLine = objectToOptional<LineString3d>(value);
-          });
+          },
+          "The stop line (LineString3d or None)");
+
+  // areas are a bit complicated because the holes are vectors of vectors. For those, we cannot rely on the automatic
```

Comment:
How about adding your test code to the tutorial (i.e. as unittest)?

Reply:
yes, I'm planning to do that. But I'll do it in a separate MR, because I wanted to rework the whole python tutorial in general.